test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTests.cs

```diff
@@ -1429,5 +1429,29 @@ public class ModelBindingTests
             Assert.Equal("test.txt", fileDetails.Filename);
             Assert.Equal("Test Content", fileDetails.Content);
         }
+
+        [Fact]
+        public async Task TryUpdateModel_ReturnDerivedAndBaseProperties()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var response = await client.GetStringAsync("http://localhost/TryUpdateModel/" +
+                "GetEmployeeAsync_BindToBaseDeclaredType" +
+                "?Parent.Name=fatherName&Parent.Parent.Name=grandFatherName&Department=Sales");
+
+            // Assert
+            var employee = JsonConvert.DeserializeObject<Employee>(response);
+
+            // Act
+            Assert.Equal("fatherName", employee.Parent.Name);
+            Assert.Equal("Sales", employee.Department);
+
+            // Includes this as there is data from value providers, the include filter
```

Comment:
nit: this what and what include filter (action doesn't filter properties)?  think you mean "Round-tripped value includes descendent instances for all properties with data in the request." but I'm not sure.